app/models/manageiq/providers/ibm_power_hmc/infra_manager/event_target_parser.rb

```diff
@@ -30,6 +30,9 @@ def parse
         # instead of querying the full LPAR data.
         $ibm_power_hmc_log.info("#{self.class}##{__method__} LPAR uuid #{uuid}")
         target_collection.add_target(:association => :vms, :manager_ref => {:ems_ref => uuid})
+      when "VirtualSwitch"
+        $ibm_power_hmc_log.info("#{self.class}##{__method__} VirtualSwitch uuid #{uuid}")
+        target_collection.add_target(:association => :host_virtual_switches, :manager_ref => {:uid_ems => uuid, :host => elems[-3]})
```

Comment:
If you want since virtual_switches are a child collection of the host you can do a targeted refresh of the whole host here:
```
target_collection.add_target(:association => :hosts, :manager_ref => {:ems_ref => elems[-3]})
```

Reply:
Thanks @agrare for your suggestion. 
I have one question about the InfraManager object.
When we do the association with `:host_virtual_switches`, I see a `manager_ref` as property here : https://github.com/ManageIQ/manageiq/blob/a0625967a209c2abce9870af72f58700c5c8abfd/app/models/manageiq/providers/inventory/persister/builder/infra_manager.rb#L164

In the case of an association with `:hosts`, I don't see the `manager_ref` collection in this : https://github.com/ManageIQ/manageiq/blob/a0625967a209c2abce9870af72f58700c5c8abfd/app/models/manageiq/providers/inventory/persister/builder/infra_manager.rb#L129

How the association works in this case ? Do we have a manager_ref property defined inside the `add_common_default_values` reference line 135 ?

Reply:
Hey @CallisteH yes `:manager_ref => [:ems_ref]` is the default, you'll only see collections overriding `:manager_ref` if it is something other than `:ems_ref`